pkg/providers/instancetype/suite_test.go

```diff
@@ -1182,6 +1181,18 @@ var _ = Describe("InstanceType Provider", func() {
 			ExpectProvisioned(ctx, env.Client, clusterNonZonal, cloudProviderNonZonal, coreProvisionerNonZonal, pod)
 			ExpectScheduled(ctx, env.Client, pod)
 
+			Expect(azureEnvNonZonal.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+			vm := azureEnvNonZonal.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
+			Expect(vm.Zones).To(BeEmpty())
+		})
+		It("should support provisioning non-zonal instance types in zonal regions", func() {
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+			pod := coretest.UnschedulablePod(coretest.PodOptions{
+				NodeSelector: map[string]string{v1.LabelInstanceTypeStable: "Standard_A0"},
```

Comment:
This `Standard_A0` is the only instance type that we currently have generated [here ](https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/fake/zz_generated.sku.eastus.go#L81) that has no zones for `eastus`.
```
LocationInfo: &[]compute.ResourceSkuLocationInfo{{Location: lo.ToPtr("eastus"), Zones: &[]string{}}},
```

However, this VM size is [restricted](https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/providers/instancetype/restricted.go#L23). Thus, it won't be provisioned. 

I tried to add `Standard_NC12s_v3` to the codegen.sh file + run it using our typical test subscription (AKS E2E - Large Cluster is what we used to use), but it shows that it has zones available in our subscription (thus this code path is not utilized).

Do we have a standard subscription (please provide name + not ID) that we use to generate these skus? I want to see what is available there that would meet these requirements. 

However, if we think it may change in the future for a particular subscription, maybe a "true" unit test would make sense just to ensure this particular path is covered.

Reply:
Wouldn't `Standard_NC6s_v3` itself work? It lists zones, but also has all of them restricted ...

Reply:
yep you're correct, that should work. I think it was end of night for me + I didn't re-check the sku.AvailabilityZones() code: https://github.com/Azure/skewer/blob/main/sku.go#L511-L563

Reply:
Updated, everything should be good now